src/IssueViz.Security/OpenInIDE/Api/OpenInIDERequestHandler.cs

```diff
@@ -55,16 +80,71 @@ Task IOpenInIDERequestHandler.ShowHotspotAsync(IShowHotspotRequest request)
                 throw new ArgumentNullException(nameof(request));
             }
 
+            return ShowHotspotAsync(request);
+        }
+
+        private async Task ShowHotspotAsync(IShowHotspotRequest request)
+        {
+            logger.WriteLine(OpenInIDEResources.OpenHotspot_ProcessingRequest, request.ServerUrl, request.ProjectKey,
+                request.OrganizationKey ?? OpenInIDEResources.OpenHotspot_NullOrganization, request.HotspotKey);
+
+            // Always show the Hotspots tool window. If we can't successfully process the
+            // request we'll show a gold bar in the window
+            toolWindowService.Show(HotspotsToolWindow.ToolWindowId);
+            failureInfoBar.Clear();
+
             if (!ideStateValidator.CanHandleOpenInIDERequest(request.ServerUrl, request.ProjectKey, request.OrganizationKey))
             {
-                return Task.CompletedTask;
+                // We're assuming the validator will have output an explanantion of why the IDE
+                // isn't in the correct state
+                ShowFailureInfoBar();
+                return;
+            }
+
+            var hotspot = await TryGetHotspotData(request.HotspotKey);
+            if (hotspot == null)
+            {
+                ShowFailureInfoBar();
+                return;
+            }
+
+            var hotspotViz = converter.Convert(hotspot);
+            if (hotspotViz == null)
+            {
+                logger.WriteLine(OpenInIDEResources.OpenHotspot_UnableToConvertHotspotData);
```

Comment:
This isn't a particularly useful message for the user. However, it will probably never occur in practice, and having the message removes any doubt about which stage the problem occurred at.